### ISSUE TITLE
[NEW FEATURE] More ways to draw debug elements than just DebugLines

### ISSUE THREAD

**Issue by bamling** [amethyst/amethyst#1651]
## Description

I'm not sure if this should be posted as a `NEW FEATURE` or a `FEATURE MODIFICATION` but seeing as `DebugLines` is *exactly* what the name says - *lines* - I figured this might be more along the lines of a new feature.

I would like to have more options when it comes to drawing debug elements/lines/shapes. Right now one can one draw `DebugLines` which works for many, if not most cases... For example, in my current project I'm utilising `DebugLines` to draw my own rectangular debug shapes, which works but could be made simpler:

```rust
// draw top line
debug_lines.draw_line(
    [x - width / 2.0, y + height / 2.0, z].into(),
    [x + width / 2.0, y + height / 2.0, z].into(),
    color,
);

// draw right line
debug_lines.draw_line(
    [x + width / 2.0, y + height / 2.0, z].into(),
    [x + width / 2.0, y - height / 2.0, z].into(),
    color,
);

// draw bottom line
debug_lines.draw_line(
    [x + width / 2.0, y - height / 2.0, z].into(),
    [x - width / 2.0, y - height / 2.0, z].into(),
    color,
);

// draw bottom line
debug_lines.draw_line(
    [x - width / 2.0, y - height / 2.0, z].into(),
    [x - width / 2.0, y + height / 2.0, z].into(),
    color,
);
```
But when it comes to circles `DebugLines` are simply not enough. Therefor I propose that more ways of drawing debug elements are added.

For a starter, **rectangles** and **circles** should be enough. Most requirements can probably met with those already.

## What problem does this solve? What need does it fill?
I'm currently using `DebugLines` to visualise where my colliders for collision detection and the likes are, but rectangles aren't exactly enough... especially when it comes to round/circular colliders (e.g. balls). This is just an example use case but I'm sure there's many others. :smiley: 

## Describe the solution you'd like
I'd like to be able to draw `DebugRectangles`/`DebugCircles` in a similar way to how `DebugLines` work. The required parameters obviously differ but the API should be similar.

**Comment by minecrawler** [comment]
It would be useful to add 3D shapes as well, mainly box and sphere. Rendering bounding-boxes should be quite standard, and it would be nice if there was some simple way to do so.

**Comment by piedoom** [comment]
Could be neat to have something like a `DebugDraw` trait that just exposes something like a `draw` method, and then implement some basic defaults while allowing users to create their own if they have more specific use cases

**Comment by Frizi** [comment]
Note that the debug lines pass specifically is only for drawing lines. If what you want is just ability to draw "wireframes" of certain shapes, you can construct them from lines, and in fact this is how such api would probably be implemented today.

**Comment by Max-Rensen** [comment]
Correct me if I'm wrong, but wouldn't it be better to simply add some methods to `DebugLinesComponent`, which add new lines to form the correct shapes. If you create a new trait or new structs for every shape, you would create a lot of overhead with new `Systems` or `Component`s that the user has to add in order to handle these objects.

**Comment by Frizi** [comment]
I think it's sufficiently implemented with #1766.
